packages/arb-token-bridge-ui/src/components/App/App.tsx

```diff
@@ -178,21 +178,27 @@ const Injector = ({ children }: { children: React.ReactNode }): JSX.Element => {
 
     const { l1, l2 } = networksAndSigners
     const isConnectedToArbitrum = isNetwork(chain.id).isArbitrum
+    const isConnectedToOrbitChain = isNetwork(chain.id).isOrbitChain
 
     const l1NetworkChainId = l1.network.id
     const l2NetworkChainId = l2.network.id
 
+    const isParentChainEthereum = isNetwork(l1NetworkChainId).isEthereum
+
     actions.app.reset(chain.id)
     actions.app.setChainIds({ l1NetworkChainId, l2NetworkChainId })
 
-    if (!isConnectedToArbitrum) {
-      console.info('Deposit mode detected:')
-      actions.app.setIsDepositMode(true)
-      actions.app.setConnectionState(ConnectionState.L1_CONNECTED)
-    } else {
+    if (
+      (isParentChainEthereum && isConnectedToArbitrum) ||
```

Comment:
Had to change this up a bit because we redefine what is deposit or withdrawal mode.

Deposit is:
- transfer from Ethereum to Arbitrum, or
- transfer from Arbitrum to Orbit

Withdrawal is:
- transfer from Arbitrum to Ethereum, or
- transfer from Orbit to Arbitrum

Notice how Arbitrum is in the middle of it, it can be either source or destination. It will cause some edge cases we will handle in this PR, so keep that in mind.